tests/builder/app_settings.rs

```diff
@@ -263,6 +263,41 @@ fn infer_subcommands_pass_conflicting_aliases() {
     assert_eq!(m.subcommand_name(), Some("test"));
 }
 
+#[test]
+fn infer_long_flag_pass_conflicting_aliases() {
+    let m = Command::new("prog")
+        .infer_subcommands(true)
+        .subcommand(Command::new("test").long_flag_aliases(["testa", "t", "testb"]))
+        .try_get_matches_from(vec!["prog", "te"])
+        .unwrap();
+    assert_eq!(m.subcommand_name(), Some("test"));
+}
+
+#[test]
+fn infer_long_flag_pass_conflicting_name() {
+    let m = Command::new("prog")
+        .infer_subcommands(true)
+        .subcommand(Command::new("test").long_flag("testa"))
+        .try_get_matches_from(vec!["prog", "te"])
```

Comment:
What is this trying to test?  It specifies a command name and a command long flag and then verifies that the command name is inferred.  This looks like any other command name inference test

Reply:
The "before state" should help and I fixed the test name. Basically with inference on and differing subcommand/long_flag names, clap crashes because it tries to find the long_flag subcommand instead of the subcommand associated with the long flag. In my last commit, notice that I changed returning the long flag name to returning the subcommand name (which fixes this crash).

Reply:
Thanks for the clarification!  Having a commit for just this with a commit message describing the bug being fixed would have been ideal.  Not a blocker though.